src/main/groovy/edu/oregonstate/mist/invencycle/resources/BikeResource.groovy

```diff
@@ -66,7 +69,60 @@ class BikeResource extends Resource {
     }
 
     /**
-     * Delete by ID
+     * POST bikes
+     */
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response postBike (@Valid Bike newBike) {
+        Response returnResponse
+        try {
+            bikeDAO.postBrakeFront(newBike.brake_make_front,
+                                   newBike.brake_model_front,
+                                   newBike.brake_rotor_size_front)
+
+            bikeDAO.postBrakeRear(newBike.brake_make_rear,
+                                  newBike.brake_model_rear,
+                                  newBike.brake_rotor_size_rear)
+            bikeDAO.postDerailuerFront(newBike.derailuer_make_front,
+                                       newBike.derailuer_model_front,
+                                       newBike.derailuer_speeds_front)
+            bikeDAO.postDerailuerRear(newBike.derailuer_make_rear,
+                                      newBike.derailuer_model_rear,
+                                      newBike.derailuer_speeds_rear)
+            bikeDAO.postFrameSize(newBike.frame_size_name,
+                                  newBike.frame_size_cm)
+            bikeDAO.postFork(newBike.fork_make,
+                             newBike.fork_model,
+                             newBike.fork_travel)
+            bikeDAO.postShock(newBike.shock_make,
+                              newBike.shock_model,
+                              newBike.shock_travel)
+            bikeDAO.postWheelFront(newBike.wheel_size_front,
+                                   newBike.rim_make_front,
+                                   newBike.rim_model_front,
+                                   newBike.hub_make_front,
+                                   newBike.hub_model_front)
+            bikeDAO.postWheelRear(newBike.wheel_size_rear,
+                                  newBike.rim_make_rear,
+                                  newBike.rim_model_rear,
+                                  newBike.hub_make_rear,
+                                  newBike.hub_model_rear)
+            bikeDAO.postBike(newBike.make,
+                             newBike.model,
+                             newBike.msrp,
+                             newBike.year,
+                             newBike.type)
+
+            returnResponse = Response.ok("Bike successfully created").build()
+        } catch (Exception e) {
+            returnResponse = internalServerError("Internal server error").build()
```

Comment:
Also, how about logging the error. You can declare this variable in your class:
Logger logger = LoggerFactory.getLogger(BikeResource.class);

Then in your catch block, you can use:
logger.error("Exception while calling: postBike", e)

This will help with your debugging.